src/core/grid_based_algorithms/lb_interface.cpp

```diff
@@ -1542,3 +1545,16 @@ Vector3d lb_lbfluid_calc_fluid_momentum() {
   return fluid_momentum;
 }
 #endif
+
+/*** Sanity check if the velocity defined at LB boundaries is within the Mach
+number limits of the scheme i.e. u < 0.3**************************************/
+
+void lb_boundary_mach_check() {
+  for (auto it = 0; it < LBBoundaries::lbboundaries.size(); ++it) {
+    for (int v = 0; v < 3; ++v) {
+      if (LBBoundaries::lbboundaries[it]->velocity()[v] > 0.3) {
```

Comment:
Shouldn't this check the absolute value? Also please check what units the boundary velocity is in, if I recall it correctly it's in MD unites, in which case a unit version would be needed here.

Reply:
both the observations are right; I did not realize that. Will correct it.